### PR TITLE
feat: populate OpenAPI objects with decorators for generated controller/model classes

### PR DESCRIPTION
**Pull request by raymondfeng**
`lb4 openapi` now generates code that keeps OpenAPI objects in corresponding decorations.

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [x] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [x] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: packages/cli/generators/openapi/utils.js [hunk 1]
```diff
@@ -182,8 +182,22 @@ function escapeComment(comment) {
   return utils.wrapText(comment, 76);
 }
 
-function toJsonStr(val) {
-  return json5.stringify(val, null, 2);
+/**
+ * Convert values to a JavaScript literal
+ * @param {*} val
+ */
+function toLiteral(val) {
```
Comment (jannyHou) [inline packages/cli/generators/openapi/utils.js:189]: could you add some unit test for it?

File: packages/cli/test/unit/openapi/controller-spec.unit.js [hunk 2]
```diff
@@ -15,75 +16,10 @@ describe('openapi to controllers/models', () => {
...
-              "return {id: customerId, 'first-name': 'John', 'last-name': 'Smith'};",
-            signatureForInterface:
-              'findCustomerById(customerId: number): Promise<Customer>',
-            signatureForNamedParams:
-              'findCustomerById(params: { customer_id: number }): Promise<Customer>',
-          },
-        ],
-      },
-    ]);
+    expectToMatchSnapshot(
```
Comment (jannyHou) [inline packages/cli/test/unit/openapi/controller-spec.unit.js:19]: hmm why have to stringify the snapshot here?
I noticed the other ones are like 
```ts
// provide the snapshot directly
expectFileToMatchSnapshot(controller);
```

Comment (raymondfeng) [reply packages/cli/test/unit/openapi/controller-spec.unit.js:19]: There are two flavors:

- Match a file to the snapshot
- Match some text to the snapshot

In controll-spec unit test, the spec is NOT written to the file system.

**Comment by gentunian** [comment]
sorry @raymondfeng ... I should have left [this comment](https://github.com/loopbackio/loopback-next/commit/ae620d4ad5e2be4bf8dcaac9b4842166b9b76215#r54333728) in this PR.

**Pull request merged**